pkg/controller/controllercmd/builder.go

```diff
@@ -163,24 +197,23 @@ func (b *ControllerBuilder) Run(config *unstructured.Unstructured, stopCh <-chan
 	}
 
 	leaderElection.Callbacks.OnStartedLeading = func(stop <-chan struct{}) {
-		if err := b.startFunc(config, clientConfig, stop); err != nil {
+		if err := b.startFunc(config, clientConfig, eventRecorder, stop); err != nil {
 			glog.Fatal(err)
 		}
 	}
 	leaderelection.RunOrDie(leaderElection)
 	return fmt.Errorf("exited")
 }
 
-func (b *ControllerBuilder) getClientConfig() (*rest.Config, error) {
-	kubeconfig := ""
-	if b.kubeAPIServerConfigFile != nil {
-		kubeconfig = *b.kubeAPIServerConfigFile
+func (b *ControllerBuilder) getNamespace() (string, error) {
+	nsBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
this will eventually need to respect the kubeconfig file, but I suspect that @enj will fix it up.

Reply:
@deads2k off topic, but I was surprised there is no method in client-go that return this...